src/main/java/controllers/AdminChangeProductServlet.java

```diff
@@ -0,0 +1,49 @@
+package controllers;
+
+import factory.ProductServiceFactory;
+import model.Product;
+import service.ProductService;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Optional;
+
+@WebServlet("/admin/changeproduct")
```

Comment:
maybe better?
```suggestion
@WebServlet("/admin/change/product")
```
the same with users